src/js/components/modals/LanguagePreferenceFormModal.js

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import { Trans } from "@lingui/macro";
+import { i18nMark } from "@lingui/react";
+import Languages from "#SRC/js/constants/Languages";
+import LanguageModalActions from "#SRC/js/events/LanguageModalActions";
+import FormModal from "../FormModal";
+import ModalHeading from "../modals/ModalHeading";
+import UserLanguageStore from "../../stores/UserLanguageStore";
+
+class LanguagePreferenceFormModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: props.isOpen || false
+    };
+
+    this.handleClose = this.handleClose.bind(this);
+    this.handleLanguagePrefSubmit = this.handleLanguagePrefSubmit.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { isOpen } = nextProps;
+    if (isOpen !== this.state.isOpen) {
+      this.setState({ isOpen });
+    }
+  }
+
+  handleClose() {
+    LanguageModalActions.close();
+    this.setState({ isOpen: false });
+  }
+
+  handleLanguagePrefSubmit(formData) {
+    UserLanguageStore.set(formData.language);
+
+    this.handleClose();
+  }
+
+  getLanguagePreferenceDefinition() {
+    const languages = Object.keys(Languages).map(lang => {
+      return { html: Languages[lang], id: lang };
+    });
+
+    return [
+      {
+        fieldType: "select",
+        label: i18nMark("Select your language"),
```

Comment:
Since there is no translation happening in the reactjs-components `Form` where this ends up getting used, I think it will need to be pretranslated using the `withI18n` HOC and `this.props.i18n._(t`Select your language`)`